### ISSUE TITLE
Custom Loss: Order of arguments

### ISSUE THREAD

**Issue by Arktius** [tensorflow/tensorflow#38067]
I've found a little mistake in the documentation. On the following website, the order of y_true and y_pred are reversed:

https://www.tensorflow.org/tutorials/customization/custom_training

```
def loss(predicted_y, target_y):
  return tf.reduce_mean(tf.square(predicted_y - target_y))
```

It's usually the other way around:

https://www.tensorflow.org/api_docs/python/tf/keras/losses/

```
keras.losses.mean_squared_error(y_true, y_pred)
```
It makes no difference for MSE since this loss is symmetric. It does make a difference for MMSE (Masked MSE) where random values of the target are mapped to zero.

**Comment by gowthamkpr** [comment]
@Arktius I don't think this is going to matter as much as we are using MSE as loss function here.

**Comment by Arktius** [comment]
That's what I've already written. But people want to know how to write their own custom loss and if the loss is not symmetric, it does matter.

**Comment by Arktius** [comment]
Code:
```
import tensorflow as tf
from tensorflow.keras.layers import Dense
from tensorflow.keras.models import Sequential
from tensorflow import math, dtypes
from tensorflow import float32 as f32 
from tensorflow.keras.optimizers import Adam
from tensorflow.keras.layers import Input
import random
import numpy as np # linear algebra

rseed=10
np.random.seed(rseed)
random.seed(rseed)
tf.compat.v1.set_random_seed(rseed)
    
def MMSE( preds,targets, mask_value=0.0):
    tf.print('\npred',preds)
    tf.print('target',targets)
    mask = dtypes.cast(tf.not_equal(targets,0),f32) 
    num_rating = math.reduce_sum(mask) #count ratings
    loss = math.reduce_sum(math.square(mask*(preds - targets))) / num_rating 
    return loss

def MMSE2( targets,preds, mask_value=0.0):
    tf.print('\npred',preds)
    tf.print('target',targets)    
    mask = dtypes.cast(tf.not_equal(targets,0),f32) 
    num_rating = math.reduce_sum(mask) #count ratings
    loss = math.reduce_sum(math.square(mask*(preds - targets))) / num_rating 
    return loss

input_dim = Input(shape = (3, ))
model = Sequential()
model.add(Dense(3,input_dim=3))
model.add(Dense(3))
model.compile(optimizer = Adam(lr=0.01),loss=[MMSE]) 
            
data  = tf.math.round(tf.random.normal(shape=[5,3]))
history = model.fit(data,data, epochs = 1, batch_size = 5,verbose=0, shuffle=False) 
print(history.history)



np.random.seed(rseed)
random.seed(rseed)
tf.compat.v1.set_random_seed(rseed)

input_dim = Input(shape = (3, ))
model = Sequential()
model.add(Dense(3,input_dim=3))
model.add(Dense(3))
model.compile(optimizer = Adam(lr=0.01),loss=[MMSE2]) 
            
data  = tf.math.round(tf.random.normal(shape=[5,3]))
history = model.fit(data,data, epochs = 1, batch_size = 5,verbose=0, shuffle=False) 


print(history.history)
```
Results:
```
pred [[0 -2 0]
 [-1 -1 1]
 [-1 -1 0]
 [1 -1 1]
 [0 -1 -1]]
target [[0.785157084 -0.0453303158 1.38228703]
 [-0.0399211645 0.472007155 1.34690034]
 [0.533699632 0.0738710314 1.13373435]
 [-0.322163254 0.278934747 0.46171847]
 [0.966199279 -0.420801282 0.477977633]]
{'loss': [1.431638240814209]}

pred [[0.785157084 -0.0453303158 1.38228703]
 [-0.0399211645 0.472007155 1.34690034]
 [0.533699632 0.0738710314 1.13373435]
 [-0.322163254 0.278934747 0.46171847]
 [0.966199279 -0.420801282 0.477977633]]
target [[0 -2 0]
 [-1 -1 1]
 [-1 -1 0]
 [1 -1 1]
 [0 -1 -1]]
{'loss': [1.5207717418670654]}
```

**Comment by ashutosh1919** [comment]
@Arktius and @MarkDaoust , please close this issue since the PR is merged.

**Comment by MarkDaoust** [comment]
Tip when submitting a PR that closes a bug.

If you put:

```
fixes: tensorflow/tensorflow#38067
```

In the commit message or PR description it will auto-close the bug on merge.
